engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -70,11 +70,27 @@ case class FixedValuesValidator(possibleValues: List[FixedExpressionValue]) exte
   }
 }
 
-case object LiteralIntValidator extends ParameterValidator {
-
+case class RegExpParameterValidator(pattern: String, message: String, description: String) extends ParameterValidator {
   override def isValid(paramName: String, value: String, label: Option[String])
                       (implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] = {
-
-    if (Try(value.toInt).isSuccess) valid(Unit) else invalid(ProcessCompilationError.InvalidLiteralIntValue(paramName, label, value))
+    if (Pattern.compile(pattern).matcher(value).matches()) valid(Unit) else invalid(NotMatchParameter(message, description, paramName, nodeId.id))
```

Comment:
Uh oh we have a bug here. I know that it was before this change, but we need to fix it. We should allow user to define `@Nullable param: Integer`. Validator shouldn't handle blank expression on BE and FE side. To make this validator composable we should handle mandatory checking only in `MandatoryValidator`. Can you write a test for that? Also I wonder what will happen in ``@Nullable param: Int` case (primitive int)

Reply:
Fix and tests: https://github.com/TouK/nussknacker/pull/816/commits/d30213d2f5cb86aa7922ab9d70beba7934200449